Source/OxyPlot/Series/StairStepSeries.cs

```diff
@@ -216,41 +222,146 @@ public override void Render(IRenderContext rc)
                     }
                 };
 
-            // Transform all points to screen coordinates
-            // Render the line when invalid points occur
+            var points = ActualPoints;
+
+            int offset = 0;
+            double xClipMax = double.MaxValue;
+
+            if (IsXMonotonic)
+            {
+                double xClipMin = XAxis.ClipMinimum;
+                xClipMax = XAxis.ClipMaximum;
+
+                WindowStartIndex = UpdateWindowStartIndex(points, point => point.X, xClipMin, WindowStartIndex);
+                offset = WindowStartIndex;
+            }
+
             var linePoints = new List<ScreenPoint>();
             var markerPoints = new List<ScreenPoint>();
-            double previousY = double.NaN;
-            foreach (var point in this.ActualPoints)
+
+            for (int i = offset; i < points.Count;)
             {
-                if (!this.IsValidPoint(point))
+                bool hasValid = FindNextValidSegment(points, i, xClipMax, out int validOffset, out int endOffset);
+                if (!hasValid)
+                    break;
+
+                ScreenPoint transformedPoint = default;
+                DataPoint previousPoint = DataPoint.Undefined;
+                bool xIncreased = false;
+                bool xDecreased = false;
+
+                for (i = validOffset; i < endOffset; ++i)
                 {
-                    renderPoints(linePoints, markerPoints);
-                    linePoints.Clear();
-                    markerPoints.Clear();
-                    previousY = double.NaN;
-                    continue;
+                    var point = points[i];
+
+                    // For performance we want to draw as few lines as possible.  Assuming sane
```

Comment:
Might want to qualify 'sane axis transforms' as 'orthogonal and monotonic'

Reply:
Yeah, so... I wasn't sure what kind of transformation it needs to be in order to make dropping points out of a straight line valid.  An affine transformation?  A linear transformation?  It's been a while since I studied linear algebra...

Reply:
As far as I can tell, it just needs to be that a change in data-X effects not change in screen-Y (and vice versa: orthogonal) and that increasing the value in data-space only increases or decreases the value in screen space (can't decrease in one place and increase in the other, monotonic). The only axes we have which violate this are those for Polar plots, which have never been supported on `StairStepSeries`.